compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -749,6 +763,11 @@ private void validateWorkerAnnAttachments(BLangExpression expr) {
 
     public void visit(BLangRecordVariable varNode) {
 
+        // Only simple variables are allowed to be configurable.
+        if (isConfigurable(varNode)) {
+            dlog.error(varNode.pos, DiagnosticErrorCode.ONLY_SIMPLE_VARIABLES_ARE_ALLOWED_TO_BE_CONFIGURABLE);
```

Comment:
Spec allows this, Can you create an issue to track this?

Reply:
Spec says `binding-pattern must be just a variable-name` so only simple variables are allowed right?

Reply:
Yes, You are correct. I missed it.